spec/support/request.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Support
+  module RequestSpec
+    class SessionHelperController < ApplicationController
+      def update
+        session[params[:key]] = params[:value]
+      end
+    end
+
+    Rails.application.routes.tap do |routes|
+      routes.disable_clear_and_finalize = true
+      routes.draw { post "__session", to: "support/request_spec/session_helper#update" }
+      routes.disable_clear_and_finalize = false
+    end
+
+    def set_session(key, value)
+      post "/__session", params: { key: key, value: value }
+    end
+
+    RSpec.configure do |rspec|
+      rspec.include self, type: :request
+    end
```

Comment:
I think in some other request specs we have a different way of using session - namely by mocking it.
```
before do
      allow_any_instance_of(Nominations::RequestNominationInviteController)
        .to receive(:session)
              .and_return(session)
    end
```
I don't mind either solution, but I want our approach to be consistent across the tests.

Reply:
I might be biased but I'd vote for the above. :P Mocking the session has a major downside of using a hash instead of a session object - they might have compatible interface, but their behaviour is slightly different. It also reads better:
```
before do
  set_session :nomination_request_form, { school_id: school.id }
end
```